adding filter by SHA for bug fixes table

y3pio opened the pull request:
## PR Summary
<!-- Short summary of the PR -->
Adding filter by SHA input option on the `bug_fixes` table.

## Details
<!-- Bullet points/details of the PR. Technical changes, implementation decision, etc... -->

- Partially address https://github.com/ponder-lab/Imperative-DL-Study-Web-App/issues/161 by adding a filter for the `but_fixes` table.
- Adding django tables filtering as suggested by their docs: https://django-tables2.readthedocs.io/en/latest/pages/filtering.html
- Does an exact match on the SHA, search with empty input to list all the entries.
- Does NOT include filtering for commits table, will need to look into this a bit more.

## Testing/QA
<!-- Details on testing/QA done for this PR. Post screenshots/caps here. -->

- Tested locally, screencap below.

https://github.com/user-attachments/assets/5c790057-cb9f-4c38-bd52-e424502d98a0

khatchad commented:
@y3pio Does this solve an issue? If so, could you please reference it in the description? Also, is this a feature, can a feature label added and the term "feat" removed from the title? Lastly, it would be good to verify this on a dev deployment instead of just locally. Thanks!

y3pio commented:
> @y3pio Does this solve an issue? If so, could you please reference it in the description? Also, is this a feature, can a feature label added and the term "feat" removed from the title? Lastly, it would be good to verify this on a dev deployment instead of just locally. Thanks!

Yup, removed the `feat` from the title and added an `enhancement` label. 

> Lastly, it would be good to verify this on a dev deployment instead of just locally. Thanks!
I think we'll need to setup the Heroku deploy again, can create an issue for this!

khatchad commented:
> > @y3pio Does this solve an issue? If so, could you please reference it in the description? Also, is this a feature, can a feature label added and the term "feat" removed from the title? Lastly, it would be good to verify this on a dev deployment instead of just locally. Thanks!
> 
> Yup, removed the `feat` from the title and added an `enhancement` label.

You can create a "feature" label. 

> > Lastly, it would be good to verify this on a dev deployment instead of just locally. Thanks!
> 
> I think we'll need to setup the Heroku deploy again, can create an issue for this!

Thanks. I think that this should be done before any other work.

y3pio commented:
Merging this to test Heroku deployment setup.
